modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsUpdateMediaCaption.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+use Drupal\media\Entity\Media;
+
+/**
+ * Configure Behavior for paragraphs.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "paragraphs_media_caption"
+ * )
+ */
+class ParagraphsUpdateMediaCaption extends ProcessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    // Get the field name from the yml.
+    $field_name = $this->configuration['field'];
+    // Get the field values.
+    $field_values = $row->getSourceProperty($field_name);
+    // Loading the media to save from the row.
+    $media = Media::load($value['target_id']);
```

Comment:
I think we'll want to add a check to make sure the media entity was loaded here.

Reply:
updated with the condition check.